spec/lib/linters/url_options_linter_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rubocop'
+require 'rubocop/rspec/support'
+require_relative '../../../lib/linters/url_options_linter'
+
+describe RuboCop::Cop::IdentityIdp::UrlOptionsLinter do
```

Comment:
a cop spec!!! neat!

Reply:
like a boss!

Reply:
Was a fun adventure!

The docs I found most helpful were here: https://github.com/rubocop-hq/rubocop/blob/master/docs/modules/ROOT/pages/development.adoc

The AST parts were the most complex for me, but referencing this existing check for whether a method is conditionally defined helped a lot: https://github.com/rubocop-hq/rubocop/blob/751edc7/lib/rubocop/cop/style/missing_respond_to_missing.rb